### PR TITLE
[TFLite] Fix the (u)int8 reference LOG_SOFTMAX operator by properly precomputing the kernel parameters

### PR DESCRIPTION
**Pull request by Tessil**
Hello,

The https://github.com/tensorflow/tensorflow/commit/6e8b672422ab9522dcc29dd8d6bfc164e7c0aade commit introduced an optimized kernel for the LOG_SOFTMAX operator using LUT but by inadvertence removed the pre-processing needed for the reference kernel while doing so. This PR restores the pre-processing and adds some tests for the reference kernel.

Thibaut

### CODE REVIEW

**Comment by abattery** [comment]
@renjie-liu @jianlijianli could you review this PR?

**Review by jianlijianli — APPROVED** [review]
Thanks Tessil for fixing the issue and adding the test cases!

**Review by renjie-liu — APPROVED** [review]

File: tensorflow/lite/kernels/activations_test.cc [hunk 1]
```diff
@@ -243,6 +247,18 @@ class LogisticOpTest : public SingleOpTest {
   }
 };
 
+const auto kLogSoftmaxKernelMap = std::map<string, TfLiteRegistration*>({
```
Comment (jdduke) [inline tensorflow/lite/kernels/activations_test.cc:250]: This violates trivially destructible static rules, please allocate this on the heap like the other map statics:

```
const auto kLogSoftmaxKernelMap = new std::map<string, TfLiteRegistration*>({...
```

Comment (Tessil) [reply tensorflow/lite/kernels/activations_test.cc]: Oh I was wondering why there was a `new` and thought it was a memory-leak mistake, I didn't know this part of the [guideline](https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables). I also fixed https://github.com/tensorflow/tensorflow/pull/48506 where I did the same.

**Review by jdduke — APPROVED** [review]

**Pull request merged**